awx/ui_next/src/screens/InstanceGroup/InstanceGroupList/InstanceGroupList.jsx

```diff
@@ -1,14 +1,209 @@
-import React from 'react';
+import React, { useEffect, useCallback } from 'react';
+import { useLocation, useRouteMatch } from 'react-router-dom';
+import { withI18n } from '@lingui/react';
+import { t } from '@lingui/macro';
 import { Card, PageSection } from '@patternfly/react-core';
 
-function InstanceGroupList() {
+import { InstanceGroupsAPI } from '../../../api';
+import { getQSConfig, parseQueryString } from '../../../util/qs';
+import useRequest, { useDeleteItems } from '../../../util/useRequest';
+import useSelected from '../../../util/useSelected';
+import PaginatedDataList, {
+  ToolbarDeleteButton,
+  ToolbarAddButton,
+} from '../../../components/PaginatedDataList';
+import ErrorDetail from '../../../components/ErrorDetail';
+import AlertModal from '../../../components/AlertModal';
+import DatalistToolbar from '../../../components/DataListToolbar';
+import InstanceGroupListItem from './InstanceGroupListItem';
+
+const QS_CONFIG = getQSConfig('instance_group', {
+  page: 1,
+  page_size: 20,
+});
+
+const modifyInstanceGroups = (items = []) => {
+  return items.map(function updateDelete(item) {
+    if (item.name === 'tower') {
+      item.summary_fields.user_capabilities.delete = false;
```

Comment:
Even though `.map()` returns a new Array, this line mutates the original instanceGroups object. We could avoid this by making a copy of the `items` array before updating it.

Reply:
Updated.